src/metabase/models/params/chain_filter.clj

```diff
@@ -445,6 +446,11 @@
             ;; FIXME: this can OOM for text column if each value are too large. See #46411
             ;; Consider using the [[field-values/distinct-text-field-rff] rff]
             values      (qp/process-query mbql-query (constantly conj))]
+        (try ; Feature issue #46888: log chain filter query.
+          (log/debugf "Chain filter native query: `%s`."
+                      (:query (qp.compile/compile mbql-query)))
```

Comment:
Do we need to recompile the query here, or would logging the native form produced by `qp/process-query` on the line above suffice?

Or else if the point is that we want debug logging of the query in case the call to `process-query` fails and throws, maybe this logging call should happen above, before the call to `process-query`?

Reply:
Recompilation. Thanks to the `(constantly conj)` argument of `process-query` call, the function returns just plain rows with no native form attached.

Point. My understanding is that users need to access sql used for _successful_ values queries, in case those values are not as what was expected. I'd expect failure of `process-query` would get logged, including native form.